### PR TITLE
feat(BoxSeries): add presence

### PR DESCRIPTION
**Pull request by Faelayis**
**Describe the changes in this pull request:**

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

reupload
image
image
image
image
image

old
image
image
image
image
image

</details>

### CODE REVIEW

File: websites/B/BoxSeries/presence.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
+const presence = new Presence({
+		clientId: "955350055034945576"
+	}),
+	strings = presence.getStrings({
+		play: "presence.playback.playing",
+		pause: "presence.playback.paused",
+		browsing: "presence.activity.browsing"
```
Comment (EncryptedDev) [inline websites/B/BoxSeries/presence.ts:7]: These strings will be removed soon, use `general.` instead. You can find the strings used in [PreMiD/Localization](https://github.com/PreMiD/Localization/blob/main/src/Presence/general.json)

File: websites/B/BoxSeries/presence.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+);
+
+presence.on("UpdateData", async () => {
+	const [Logo, Cover, Privacy, Buttons, Timestamps] = await Promise.all([
+			presence.getSetting<number>("logo"),
+			presence.getSetting<boolean>("cover"),
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("buttons"),
+			presence.getSetting<boolean>("timestamps")
+		]),
```
Comment (EncryptedDev) [inline websites/B/BoxSeries/presence.ts:31]: We prefer the use of camelCase
```suggestion
	const [logo, cover, privacy, buttons, timestamps] = await Promise.all([
			presence.getSetting<number>("logo"),
			presence.getSetting<boolean>("cover"),
			presence.getSetting<boolean>("privacy"),
			presence.getSetting<boolean>("buttons"),
			presence.getSetting<boolean>("timestamps")
		]),
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Heya! 👋

Thank you for contributing to PreMiD/Presences. Please address my review found below:

File: websites/B/BoxSeries/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	"settings": [
+		{
+			"id": "privacy",
+			"title": "Privacy Mode",
+			"icon": "fad fa-user-secret",
+			"value": false
+		},
+		{
+			"id": "timestamps",
+			"title": "Show timestamps",
```
Comment (Slowlife01) [inline websites/B/BoxSeries/dist/metadata.json:32]:
```suggestion
			"title": "Show Timestamps",
```

Consistency

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

File: websites/B/BoxSeries/presence.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+				?.textContent ?? "pesquisa não encontrada",
+		titleEp =
+			document.querySelector("#single > h1.epih1")?.textContent.split(" - ") ??
+			"pesquisa não encontrada",
+		pathArray = document.location.toString().split("/"),
+		presenceData: PresenceData = {
+			largeImageKey: ["default", "default_horizontal_tp"][logo],
+			startTimestamp: browsingTimestamp
+		};
+	if (location.search.includes("s")) {
```
Comment (Bas950) [inline websites/B/BoxSeries/presence.ts:43]: probs using [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) is better

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — DISMISSED** [review]

File: websites/B/BoxSeries/iframe.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const iframe = new iFrame();
+
+iframe.on("UpdateData", async () => {
+	const video: HTMLVideoElement =
+		document.querySelector(".jw-video") ||
+		document.querySelector(".html5-video-container");
+	if (video && !isNaN(video.duration)) {
```
Comment (EncryptedDev) [inline websites/B/BoxSeries/iframe.ts:7]:
```suggestion
	if (!isNaN(video?.duration)) {
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/B/BoxSeries/presence.ts [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+		}`;
+		presenceData.smallImageKey = "search";
+	} else {
+		switch (pathArray[3]) {
+			case "calendario":
+				presenceData.details = "Vendo Calendário do Dia";
+				presenceData.smallImageKey = "reading";
+				break;
+			case "generos":
+				presenceData.details = "Vendo Gêneros";
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:58]:
```suggestion
				presenceData.details = "Vendo gêneros";
```

File: websites/B/BoxSeries/presence.ts [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+				if (!privacy && pathArray[4]) presenceData.state = titleMain;
+				presenceData.smallImageKey = "reading";
+				break;
+			case "anos":
+				presenceData.details = "Vendo anos";
+				presenceData.smallImageKey = "reading";
+				break;
+			case "release":
+				presenceData.details = "Vendo Liberação";
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:67]: Não sei até que ponto é que isto é correto no brasil porque soa estranho mas pronto
```suggestion
				presenceData.details = "Vendo liberação";
```

File: websites/B/BoxSeries/presence.ts [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+				if (!privacy && pathArray[4]) presenceData.state = titleMain;
+				presenceData.smallImageKey = "reading";
+				break;
+			case "series":
+				presenceData.details = "Vendo Séries";
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:72]:
```suggestion
				presenceData.details = "Vendo séries";
```

File: websites/B/BoxSeries/presence.ts [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+				if (!video.paused) {
+					[, presenceData.endTimestamp] = presence.getTimestamps(
+						Math.floor(video.current),
+						Math.floor(video.duration)
+					);
+				}
+				if (buttons) {
+					presenceData.buttons = [
+						{
+							label: "Assistir episodio",
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:107]:
```suggestion
							label: "Assistir episódio",
```

File: websites/B/BoxSeries/presence.ts [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+					[, presenceData.endTimestamp] = presence.getTimestamps(
+						Math.floor(video.current),
+						Math.floor(video.duration)
+					);
+				}
+				if (buttons) {
+					presenceData.buttons = [
+						{
+							label: "Assistir filmes",
+							url: document.location.href.replace(/#\d+/, "")
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:139]: porquê filmes no plural?

File: websites/B/BoxSeries/presence.ts [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+						}
+					];
+				}
+				break;
+			default:
+				presenceData.details = "Vendo página";
+				presenceData.state = "em Página Inicial";
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:146]: isto não faz sentido

File: websites/B/BoxSeries/presence.ts [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+				presenceData.smallImageKey = "reading";
+				break;
+		}
+	}
+	if (!buttons) delete presenceData.buttons;
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:151]: nas linhas acima só definiste os botões se buttons fosse truthy, isto não é necessário

**Review by ImRodry — CHANGES_REQUESTED** [review]

**Comment by ImRodry** [comment]
Just as a note about commitlint, the perf scope means you changed something about performance, which you didn't in the last two commits. You should be using fix for fixes, feat for new features, refactor for code refactors with no impact on the end result and chore for non-code related changes

File: websites/B/BoxSeries/dist/metadata.json [hunk 13]
```diff
@@ -0,0 +1,59 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.6",
+	"author": {
+		"name": "Faelayis",
+		"id": "328731868096888833"
+	},
+	"service": "BoxSeries",
+	"description": {
+		"en": "Series & Movies Online in HD, choose your favorite series or movie and watch dubbed or subtitled! Join now it's free!",
```
Comment (ImRodry) [inline websites/B/BoxSeries/dist/metadata.json:9]: where did you take this from? Movies and Online shouldn't be capitalized here either

File: websites/B/BoxSeries/presence.ts [hunk 14]
```diff
@@ -0,0 +1,160 @@
+const presence = new Presence({
+		clientId: "955350055034945576"
+	}),
+	strings = presence.getStrings({
+		play: "general.playing",
+		pause: "general.paused",
+		browsing: "general.browsing"
+	}),
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:8]: why use 3 strings if the rest is in a hardcoded language? You can just remove this IMO

File: websites/B/BoxSeries/presence.ts [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+				?.textContent ?? "Pesquisa não encontrada",
+		titleEp =
+			document.querySelector("#single > h1.epih1")?.textContent.split(" - ") ??
+			"Pesquisa não encontrada",
+		pathArray = document.location.toString().split("/"),
+		presenceData: PresenceData = {
+			largeImageKey: ["default", "default_horizontal_tp"][logo],
+			startTimestamp: browsingTimestamp
+		};
+	if (new URLSearchParams(location.search).has("s")) {
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:43]: good heavens no
```suggestion
	if (location.searchParams.has("s")) {
```

File: websites/B/BoxSeries/presence.ts [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+		presenceData.details = "Pesquisando po";
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:44]:
```suggestion
		presenceData.details = "Pesquisando por";
```

File: websites/B/BoxSeries/presence.ts [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+				presenceData.details = "Vendo séries";
+				if (cover) {
+					presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+						"div > div > div > div.poster-season > img"
+					).src;
+				}
+				if (!privacy && pathArray[4]) {
+					presenceData.state = `${
+						document.querySelector("div > div > div > div.data > h1")
+							?.textContent ?? "pesquisa não encontrada"
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:81]:
```suggestion
							?.textContent ?? "Pesquisa não encontrada"
```

File: websites/B/BoxSeries/presence.ts [hunk 18]
```diff
@@ -0,0 +1,160 @@
...
+				if (!video.paused) {
+					[, presenceData.endTimestamp] = presence.getTimestamps(
+						Math.floor(video.current),
+						Math.floor(video.duration)
+					);
+				}
+				if (buttons) {
+					presenceData.buttons = [
+						{
+							label: "Assistindo episódio",
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:107]: Lembra-te de que isto é um botão
```suggestion
							label: "Assistir episódio",
```

File: websites/B/BoxSeries/presence.ts [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+				if (cover) {
+					presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+						"#dt_contenedor > div.season-bg-bs > div > img"
+					).src;
+				}
+				presenceData.details = "Vendo filmes";
+				presenceData.state = `${
+					document.querySelector(
+						"#single > div.content.full_width_layout > div.dooplay_player > div.area-season.movie-bs > div.data > h1"
+					)?.textContent ?? "pesquisa não encontrada"
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:123]:
```suggestion
					)?.textContent ?? "Pesquisa não encontrada"
```

File: websites/B/BoxSeries/presence.ts [hunk 20]
```diff
@@ -0,0 +1,160 @@
...
+				if (!video.paused) {
+					[, presenceData.endTimestamp] = presence.getTimestamps(
+						Math.floor(video.current),
+						Math.floor(video.duration)
+					);
+				}
+				if (buttons) {
+					presenceData.buttons = [
+						{
+							label: "Assistindo filmes",
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:138]:
```suggestion
							label: "Assistir filmes",
```

File: websites/B/BoxSeries/presence.ts [hunk 21]
```diff
@@ -0,0 +1,160 @@
...
+							url: document.location.href.replace(/#\d+/, "")
+						}
+					];
+				}
+				break;
+			default:
+				presenceData.details = "página inicial";
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:145]:
```suggestion
				presenceData.details = "Página inicial";
```

File: websites/B/BoxSeries/presence.ts [hunk 22]
```diff
@@ -0,0 +1,160 @@
...
+				presenceData.smallImageKey = "reading";
+				break;
+		}
+	}
+	if (!timestamps) {
+		delete presenceData.startTimestamp;
+		delete presenceData.endTimestamp;
+	}
+	if (privacy) {
```
Comment (ImRodry) [inline websites/B/BoxSeries/presence.ts:154]: there are multiple `if (!privacy)` checks up above, please check those and adjust the code accordingly to remove redundancy. Also I don't think buttons should be disabled if privacy is disabled? If you do then the setting should have an if statement in the metadata file

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/B/BoxSeries/presence.ts (hunk 15 above)
Comment (Faelayis) [reply websites/B/BoxSeries/presence.ts:37]: It shows `Property 'searchParams' does not exist on type 'Location'.ts(2339)`

Comment (ImRodry) [reply websites/B/BoxSeries/presence.ts:37]: ignore that then, confused this for something else

**Comment by ImRodry** [comment]
I told you to disregard that comment, undo the last commit

File: websites/B/BoxSeries/dist/metadata.json [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+			"icon": "fad fa-user-secret",
+			"value": false
+		},
+		{
+			"id": "timestamps",
+			"title": "Show Timestamps",
+			"icon": "fad fa-stopwatch",
+			"value": true
+		},
+		{
```
Comment (EncryptedDev) [inline websites/B/BoxSeries/dist/metadata.json:36]: Maybe add if privacy setting is true, buttons are disabled

File: websites/B/BoxSeries/dist/metadata.json [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+			"id": "logo",
+			"title": "Icon",
+			"icon": "fad fa-images",
+			"value": 0,
+			"values": [
+				"Normal",
+				"Horizontal"
+			]
+		},
+		{
```
Comment (EncryptedDev) [inline websites/B/BoxSeries/dist/metadata.json:52]: Maybe add if privacy setting is true, cover image is disabled

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — DISMISSED** [review]

**Review by Slowlife01 — CHANGES_REQUESTED** [review]
When in privacy mode you hide the settings but don't do anything with them?

**Review by Slowlife01 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**